### ISSUE TITLE
jwt_get_grant_int returns -1 when grant does not exist

### ISSUE THREAD

**Issue by nplus11** [benmcollins/libjwt#36]
***Bug***

Calling `jwt_get_grant_int` with a key that does not exist will return `-1`.

***Explanation***

This is caused by the [`get_jst_int` function](https://github.com/benmcollins/libjwt/blob/master/libjwt/jwt.c#L240) added in [cd227d7](https://github.com/benmcollins/libjwt/commit/cd227d70c497350a56c1759999a08455f5a55612). The underlying call to jansson `json_object_get` returns `NULL` when the provided key is not found in the hash maps. Upon receiving `NULL`, `get_jst_int` returns the initial set `val` of `-1` which is directly returned by `jwt_get_grant_int`.

***Problems***

1. This violates [the documentation that states `jwt_get_grant_int` will return 0 if not found](https://github.com/benmcollins/libjwt/blob/master/include/jwt.h#L157).

***Solution***

1. Adding a check to `jwt_get_grant_int` is the least invasive:
    ```c
    long jwt_get_grant_int(jwt_t *jwt, const char *grant)
    {
            ...
            long ret = get_js_int(jwt->grants, grant);
            return ((ret == -1) ? 0 : ret);
    }
    ```
1. An alternate solution would be setting the ret val initially to `0` in `get_js_int`.
    ```c
    static long get_js_int(json_t *js, const char *key)
    {
	    long val = 0;
            ...
    }
    ```

However I did not dive too deep. I'm not sure of the sure of the repercussions of either solution, so I'll let you determine the best course of action. Just wanted to let you know.

**Comment by benmcollins** [comment]
Thanks for catching this. In retrospect, I would have preferred that `jwt_get_grant_int()` had taken a `long *` argument and returned 0 or EEXIST based on the grant existing, thus alleviating any ambiguity.

Perhaps I can still salvage this and (using your fix) make the function set `errno = EEXIST` as well as returning 0 when the grant is not there.

**Comment by hamano** [comment]
I think it's better set errno = ENOENT when not found. fix #49

**Comment by benmcollins** [comment]
This now sets errno to ENOENT when the grant doesn't exist.
